### PR TITLE
feat: history server implementation

### LINKED ISSUE #1: feat: launch
- [ ] Launch service
- [ ] Implement basic API
- [ ] Add DocumentDB
- [ ] Add Docker Compose Local setup

### PR DESCRIPTION
**Pull request by xav**
# Description

Add history server implementation

Resolves #1

## How Has This Been Tested?

- Integration tests against MongoDB
- Functional tests for message and registration endpoints

### CODE REVIEW

File: src/auth/jwt.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        domain::{ClientId, DecodedClientId},
+    },
+    std::collections::HashSet,
+};
+
+#[derive(Debug)]
+pub struct Jwt(pub String);
+
+impl Jwt {
+    pub fn decode(&self, aud: &HashSet<String>) -> Result<ClientId, JwtVerificationError> {
```
Comment (HarryET) [inline src/auth/jwt.rs:21]: I think this was put into the walletconnect rust lib

Comment (xav) [reply src/auth/jwt.rs:21]: Is it? Will take a look

Comment (xav) [reply src/auth/jwt.rs]: Thanks :)

**Review by arein — APPROVED** [review]
Nice work. I mainly looked through the nice functional HTTP and store tests as the actual implementation was a bit too large to review honestly.

But based on the tests it looks very solid!

**Pull request merged**